backend/app/endpoints/squads.py

```diff
@@ -1,170 +1,235 @@
+        return {"message": "User joined the squad"}
+
+    squad_repository.invite_user(squad, user)
+
+    return {"message": "User invited"}
+
+
+@router.put("/{squad_id}/invites")
+def update_invite(
+    payload: SquadReplyRequest,
+    squad: Squad = Depends(get_squad),
+    current_user: User = Depends(get_current_user),
     squad_repository: SquadRepository = Depends(),
 ):
-    squad = squad_repository.get_squad(id)
+    if current_user not in squad.invitations:
+        raise HTTPException(status_code=400, detail="User is not invited")
 
-    if not squad:
-        raise HTTPException(status_code=404, detail="Squad not found")
+    if payload.accept:
+        squad_repository.accept_user(squad, current_user)
+    else:
+        squad_repository.decline_user(squad, current_user)
 
-    return squad.invitations
+    return {"message": "Invitation updated"}
 
 
-@router.delete("/{id}")
-def delete_squad(
-    id: int,
+@router.put("/{squad_id}/applies/{user_id}")
+def update_application(
+    user_id: int,
+    payload: SquadReplyRequest,
+    squad: Squad = Depends(get_owned_squad),
+    squad_repository: SquadRepository = Depends(),
+    user_repository: UserRepository = Depends(),
+):
+    user = user_repository.get_by_id(user_id)
+    if not user:
+        raise HTTPException(status_code=404, detail="User not found")
+
+    if user not in squad.applications:
+        raise HTTPException(status_code=400, detail="User didn't apply to this squad")
+
+    if payload.accept:
+        squad_repository.accept_user(squad, user)
+    else:
+        squad_repository.decline_user(squad, user)
+
+    return {"message": "Application updated"}
+
+
+@router.put("/{squad_id}/apply")
+def apply_to_squad(
+    squad: Squad = Depends(get_squad),
     current_user: User = Depends(get_current_user),
     squad_repository: SquadRepository = Depends(),
 ):
-    squad = squad_repository.get_squad(id)
+    if current_user in squad.members:
+        raise HTTPException(status_code=400, detail="User is already in squad")
 
-    if not squad:
-        raise HTTPException(status_code=404, detail="Squad not found")
+    if current_user in squad.invitations:
+        # User is already invited, should be automatically accepted
+        squad_repository.accept_user(squad, current_user)
+        return {"message": "User joined the squad"}
+
+    squad_repository.apply_to_squad(squad, current_user)
+
+    return {"message": "Application created"}
 
-    if squad.owner_id != current_user.id:
-        raise HTTPException(status_code=403, detail="You are not the owner")
 
+@router.get("/{squad_id}/invites", response_model=list[UserResponse])
+def get_invited_users(
+    squad: Squad = Depends(get_squad),
+):
+    return squad.invitations
+
+
+@router.get("/{squad_id}/applies", response_model=list[UserResponse])
+def get_squad_applications(
+    squad: Squad = Depends(get_squad),
+):
+    return squad.applications
+
+
+@router.delete("/{squad_id}")
```

Comment:
Add the resonse model:
```suggestion
@router.delete("/{squad_id}", response_model=...)
```

Reply:
delete should not have a response, changing it to 204.